### PR TITLE
SDE-2544: Allow setting AWS EC2Metadata in machine config pool

### PR DESCRIPTION
**Pull request by tsorya**
[SDE-2544](https://issues.redhat.com//browse/SDE-2544): Allow setting AWS EC2Metadata in machine config pool

[HIVE-2211](https://issues.redhat.com//browse/HIVE-2211)

### CODE REVIEW

File: apis/hive/v1/aws/machinepool.go [hunk 1]
```diff
@@ -21,6 +21,10 @@ type MachinePoolPlatform struct {
 	// SpotMarketOptions allows users to configure instances to be run using AWS Spot instances.
 	// +optional
 	SpotMarketOptions *SpotMarketOptions `json:"spotMarketOptions,omitempty"`
+
+	// EC2MetadataOptions defines metadata service interaction options for EC2 instances in the machine pool.
+	// +optional
+	EC2Metadata *EC2Metadata `json:"metadataService,omitempty"`
```
Comment (2uasimojo) [inline apis/hive/v1/aws/machinepool.go:27]: Weird json name, but I see you're just matching the one in the installer type :shrug:

**Review by 2uasimojo — COMMENTED** [review]
This looks great @tsorya, thanks!
- Please create a card in the HIVE project
- I think you'll need a `make update` to regenerate that saas template (`verify` failure)

The e2e-pool failure is a flake. (I don't expect this change to affect that test.)

File: apis/hive/v1/aws/machinepool.go (hunk 1 above)
Comment (tsorya) [reply apis/hive/v1/aws/machinepool.go:27]: Yeah, not mine.  I think better to match to installer's

**Review by 2uasimojo — COMMENTED** [review]
/lgtm

@abutcher FYI

**Pull request merged**